### PR TITLE
Read key format config value at runtime

### PR DESCRIPTION
**Pull request by Frost**
Previously, the `:key_format` config value was read into a module attribute
`@key_formatter` in `JaSerializer.Formatter.Utils`.

It was therefore read from the config at compile-time and stored in the compiled
module.

This leads to two things:

1. Tests using `Application.put_env/3` don't work as expected
   The tests that use `Application.put_env/3` don't work as one might think they
   do, since while the `put_env` call does change the config value, it does not
   change the already compiled `@key_formatter` module attribute, which means
   that all the tests will keep using the default key formatting, regardless of
   the config value.

2. The library has to be recompiled after changing `key_format` config
   The library has to be recompiled at any time the `key_format` config key is
   changed, which has led to issues #63 and #261. #170, which was followed up by
   #172, fixes a similar issue.

I found [this article on erlang-solutions.com](https://www.erlang-solutions.com/blog/elixir-module-attributes-alchemy-101-part-1.html)
 that talks a bit more about module attributes and contains this nice piece of advice:

> The takeaway from this is: when using module attributes, always be aware of
> what they evaluate to, making a wrong assumption can result in having to
> recompile your project.

One other thing I fix here is that the custom key formatter set in one test in
included_tests.exs, `{:custom, Macro, nil, :underscore}` does not work, as it
ends up trying to call `Macro.nil`, which does not exist. Instead, I replaced
this with `{:custom, String, :capitalize, :downcase}` and changed any failing
tests accordingly.

### CODE REVIEW

**Comment by Frost** [comment]
TL;DR on this: These changes remove the need to run `mix deps.clean ja_serializer && mix deps.get` after changing the `key_format` config.

File: test/ja_serializer/builder/included_test.exs [hunk 1]
```diff
@@ -306,13 +306,13 @@ defmodule JaSerializer.Builder.IncludedTest do
     assert "p1" in ids
 
     Application.put_env(:ja_serializer, :key_format, :underscored)
-    json = JaSerializer.format(ArticleSerializer, a1, %{}, include: "author.publishing-agent")
+    json = JaSerializer.format(ArticleSerializer, a1, %{}, include: "author.publishing_agent")
     ids = Enum.map(json["included"], &(Map.get(&1, "id")))
-    assert not "p1" in ids
```
Comment (beerlington) [inline test/ja_serializer/builder/included_test.exs]: Why is this assertion changing?

Comment (Frost) [reply test/ja_serializer/builder/included_test.exs]: That is because I also changed the string `publishing-agent` to `publishing_agent` two lines above, in order to make a positive assertion to prove that the `:underscored` key format also works.

The same goes for the assertion a few lines down as well.

It might make sense to split this test into multiple different test cases in a `describe "multi-word relationship path keys"` block, and then also have both positive and negative assertions for both `:dasherized` and `:underscored` key formats.

**Comment by Frost** [comment]
@beerlington This one is also rebased now, and I made sure to run `mix format` on the changed files again.
